foreshadow/preprocessor.py

```diff
@@ -338,19 +338,20 @@ def serialize(self):
 
         """
         json_cols = {
-            k: (
-                self._intent_map[k].__name__,
-                serialize_pipeline(
+            k: {
+                "intent": self._intent_map[k].__name__,
```

Comment:
As a common best practice, we might want to define some CONSTANTS for these keywords since we have them all over the place. No need to change them right now though.